samza-core/src/main/java/org/apache/samza/config/SystemConfig.java

```diff
@@ -106,6 +123,19 @@ public SystemAdmin getSystemAdmin(String systemName) {
     return getSystemAdmins().get(systemName);
   }
 
+  /**
+   * Get {@link SystemAdmin} instance for given system name with an adminLabel.
+   * This function provides an extra input parameter to {@link #getSystemAdmin}, which can be used to provide extra
+   * information for the admin instance, e.g. ownership of client instance, to help better identify admins in logs,
+   * threads and client instances etc., along with other relevant information like systemName.
+   *
+   * @param systemName System name
+   * @return SystemAdmin of the system if it exists, otherwise null.
+   */
+  public SystemAdmin getSystemAdmin(String systemName, String adminLabel) {
+    return getSystemAdmins(adminLabel).get(systemName);
```

Comment:
I understand this is an existing code, but it seems weird that we create ALL admins EVERY time just to get a single one.
Either we should create a single Admin, or we can cache the results of the previous getAdmins().